### ISSUE TITLE
bug: Incorrectly transforming directory paths (e.g,, ./test.md)

### ISSUE THREAD

**Issue by techfg** [vernak2539/astro-rehype-relative-markdown-links#12]
When a url is pointing to directory rather than a file, the url is being transformed.

For example, in a directory structure such as
```text
.
├── src/
│   ├── content/
│   │   ├── posts/
│   │   |   ├── archive.md/
│   │   |   |   └── index.md
│   │   |   └── post-1.md
```
  
The following markdown in `src/content/posts/post-1.md` strips `archive.md`
```markdown
Markdown: [Archive Index](./archive.md)
Transformed url: `/posts/archive`
```

Repro:
- <https://stackblitz.com/edit/withastro-astro-7ge5gz>

With Proposed Fixes from PR #21:
- <https://stackblitz.com/edit/withastro-astro-mtqt5k>

Steps to reproduce:
1. Open repro
2. Click on `Post 1` link
3. Click on either link under `Repro for Issue #12`

Expected Result:
Link should navigate to `/posts/archive.md` and show the index page

Actual Result: 
`md` is stripped and link results in 404

Additional Info:
You can see all links that Astro generates on home page under `Issue #12` heading

**Comment by techfg** [comment]
FYI - Edited OP to a [Stackblitz](https://stackblitz.com/edit/github-pdzrjo-qnmg8l) to the repro that contains the proposed fixes from #21.

**Comment by vernak2539** [comment]
Released in [v0.8.1](https://www.npmjs.com/package/astro-rehype-relative-markdown-links/v/0.8.1). Thanks!

**Comment by techfg** [comment]
Awesome, ty!
